internal/auth/authz.go

```diff
@@ -0,0 +1,77 @@
+package auth
+
+import (
+	"context"
+	"strings"
+
+	"github.com/kopia/kopia/repo"
+)
+
+// AuthorizerFunc gets the authorizations for given user.
+type AuthorizerFunc func(ctx context.Context, rep repo.Repository, username string) AuthorizationInfo
+
+// AccessLevel specifies access level when accessing repository objects.
+type AccessLevel int
+
+// Access levels.
+const (
+	AccessLevelNone   AccessLevel = iota
+	AccessLevelRead               // RO access
+	AccessLevelAppend             // RO + create new
+	AccessLevelFull               // read/write/delete
+)
+
+// AuthorizationInfo determines logged in user's access level.
+type AuthorizationInfo interface {
+	// ContentAccessLevel determines whether the user can read/write contents.
+	ContentAccessLevel() AccessLevel
+
+	// ManifestAccessLevel determines whether the user has access to a manifest with given labels.
+	ManifestAccessLevel(labels map[string]string) AccessLevel
+}
+
+type noAccessAuthorizer struct{}
+
+func (noAccessAuthorizer) ContentAccessLevel() AccessLevel { return AccessLevelNone }
+func (noAccessAuthorizer) ManifestAccessLevel(labels map[string]string) AccessLevel {
+	return AccessLevelNone
+}
+
+// NoAccess implements AuthorizationInfo which grants no permissions.
+var NoAccess AuthorizationInfo = noAccessAuthorizer{}
```

Comment:
nit: should this be a function to prevent accidental overwriting of the variable? Essentially, have it behave more like a constant.

```suggestion
func NoAccess() AuthorizationInfo { return noAccessAuthorizer{} }
```